instrumentor/controllers/instrumentationdevice/pods_webhook.go

```diff
@@ -79,19 +80,33 @@ func (p *PodsWebhook) injectOdigosEnvVars(ctx context.Context, logger logr.Logge
 			return fmt.Errorf("namespace is empty for pod %s/%s, Skipping Injection of ODIGOS environment variables", pod.Namespace, pod.Name)
 		}
 	}
+	var workloadInstrumentationConfig odigosv1.InstrumentationConfig
+	instrumentationConfigName := workload.CalculateWorkloadRuntimeObjectName(podWorkload.Name, podWorkload.Kind)
+
+	if err := p.Get(ctx, client.ObjectKey{Namespace: podWorkload.Namespace, Name: instrumentationConfigName}, &workloadInstrumentationConfig); err != nil {
+		return fmt.Errorf("failed to get instrumentationConfig: %w", err)
+	}
+	runtimeDetails := workloadInstrumentationConfig.Status.RuntimeDetailsByContainer
 
 	var serviceName *string
 	var serviceNameEnv *corev1.EnvVar
 
 	for i := range pod.Spec.Containers {
 		container := &pod.Spec.Containers[i]
+		pl := getLanguageOfContainer(runtimeDetails, container.Name)
+		if pl == common.UnknownProgrammingLanguage {
+			fmt.Println("Skipping container as programming language is unknown")
+			continue
+		}
 
-		pl, otelsdk, found := containerutils.GetLanguageAndOtelSdk(container)
+		otelSdk, found := sdks.GetDefaultSDKs()[pl]
 		if !found {
+			fmt.Println("No default SDK found for language", pl)
 			continue
 		}
 
-		webhookenvinjector.InjectOdigosAgentEnvVars(ctx, p.Client, logger, *podWorkload, container, pl, otelsdk)
+		webhookdeviceinjector.InjectOdigosInstrumentationDevice(ctx, p.Client, logger, *podWorkload, container, pl, otelSdk)
```

Comment:
this function name is `injectOdigosEnvVars`.
consider moving this logic to a different function with good name